File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet <Character> set = new HashSet<>();

        int l = 0;
        int r = 0;
        int len = 0;
       
        while(r<s.length()){
            char curr = s.charAt(r);
            if(!set.contains(curr)){
                set.add(curr);
                len = Math.max(len,r-l+1);
                r++;
            }else{
                set.remove(s.charAt(l));
                l++;
            }
        }

        return len;
        
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let set = new Set()
    let left = 0 
    let maxLength = 0

    for(let curr=0; curr<s.length; curr++){

        while(set.has(s[curr])){
            set.delete(s[left])
            left++
        }

        set.add(s[curr])
        maxLength = Math.max(maxLength, (curr+1) -left)
    }
    return maxLength
};